test/tests/rackhd11/test_rackhd11_api_templates.py

```diff
@@ -26,17 +29,13 @@ def test_api_11_templates_library(self):
                     print "Checking:", item['name'], subitem
                 self.assertGreater(len(item[subitem]), 0, subitem + ' field error')
 
-    def test_api_11_templates_library_ID(self):
-        api_data = fit_common.rackhdapi("/api/1.1/templates/library")
-        self.assertEqual(api_data['status'], 200, "Was expecting code 200. Got " + str(api_data['status']))
-        for item in api_data['json']:
-            lib_data = fit_common.rackhdapi("/api/1.1/templates/library/" + item['name'])
-            self.assertEqual(lib_data['status'], 200, "Was expecting code 200. Got " + str(lib_data['status']))
-            # check required fields
-            for subitem in ['contents', 'createdAt', 'id', 'name', 'updatedAt']:
-                if fit_common.VERBOSITY >= 2:
-                    print "Checking:", item['name'], subitem
-                self.assertGreater(len(item[subitem]), 0, subitem + ' field error')
+    def test_api_11_templates_library_ID_put_get_delete(self):
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid?scope=global", action="text-put", payload="null")
+            self.assertEqual(api_data['status'], 200, "Was expecting code 200. Got " + str(api_data['status']))
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid")
+            self.assertIn(api_data['json']['contents'], "null", "Data 'null' was not returned.")
+            self.assertEqual(api_data['status'], 200, "Was expecting code 200. Got " + str(api_data['status']))
+            api_data = fit_common.rackhdapi("/api/1.1/templates/library/testid", action="delete")
```

Comment:
Should have a check after the delete to make sure that it returned the expected status code and that the item actually went away.